### PR TITLE
make caller info sample work with C#

### PR DESCRIPTION
**Pull request by CaringDev**
## Summary
Make this sample work when called from C#

### TL;DR;
As it is, C# (respectively the C# compiler) does not understand how to handle e.g. parameters of type `option<string>` decorated with some `[CallerInfoX]` attribute. Instead we would need to supply some `FSharpOption<string>`. Marking the `option`al parameter `[<Optional>]` leads to an optional option, which is OK from C# perspective<sup>1</sup> but always `null`, i.e. `None`. I believe, we should have examples that work for F# and C# out-of-the-box. I'm happy to provide two samples including an explanation, if you think different.

<sup>1</sup> maybe the compiler should emit a warning if it encounters a [CallerInfo] attribute on a type it cannot supply?

### CODE REVIEW

File: docs/fsharp/language-reference/caller-information.md [hunk 1]
```diff
@@ -22,24 +22,22 @@ The following example shows how you might use these attributes to trace a caller
...
 open System.Diagnostics
 open System.Runtime.CompilerServices
+open System.Runtime.InteropServices
 
 type Tracer() =
     member __.DoTrace(message: string,
-                      [<CallerMemberName>] ?memberName: string,
-                      [<CallerFilePath>] ?path: string,
-                      [<CallerLineNumber>] ?line: int) =
+                      [<CallerMemberName; Optional>] memberName: string,
```
Comment (cartermp) [inline docs/fsharp/language-reference/caller-information.md]:
```suggestion
                      [<CallerMemberName; Optional; DefaultParameterValue("")>] memberName: string,
```

File: docs/fsharp/language-reference/caller-information.md [hunk 2]
```diff
@@ -22,24 +22,22 @@ The following example shows how you might use these attributes to trace a caller
...
+                      [<CallerFilePath; Optional>] path: string,
```
Comment (cartermp) [inline docs/fsharp/language-reference/caller-information.md]:
```suggestion
                      [<CallerFilePath; Optional; DefaultParameterValue("")>] path: string,
```

File: docs/fsharp/language-reference/caller-information.md [hunk 3]
```diff
@@ -22,24 +22,22 @@ The following example shows how you might use these attributes to trace a caller
...
+                      [<CallerLineNumber; Optional>] line: int) =
```
Comment (cartermp) [inline docs/fsharp/language-reference/caller-information.md]:
```suggestion
                      [<CallerLineNumber; Optional; DefaultParameterValue(0)>] line: int) =
```

**Review by cartermp — CHANGES_REQUESTED** [review]
Looks good modulo these changes (this would make the sample equivalent to the C# docs)

**Comment by CaringDev** [comment]
@cartermp ah, sorry, I should have batch-merged your suggestions instead of fixing them locally... now Github does not recognise them as resolved. Could you check again? Thank you.

**Review by cartermp — APPROVED** [review]
Thanks @CaringDev
